pkg/controller/namespace/namespace_controller.go

```diff
@@ -53,6 +64,11 @@ type ReconcileNamespace struct {
 	// This client, initialized using mgr.Client() above, is a split client
 	// that reads objects from the cache and writes to the apiserver
 	client client.Client
+
+	// this avoid the cache, which we need to bypass because the default client will attempt to place
+	// a watch on Namespace at cluster scope, which isn't desirable to us...
+	rClient client.Reader
```

Comment:
Could we have a client which would delegate to a proper instance rather than passing along two different interfaces?

Reply:
We don't want to use this client for all "Get" or "List", as we certainly do want watches and caches, we just don't want them for a couple of very specific Get/List operations. As such, not sure there's a generic to define what's the "proper" instance.